flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/DefaultDeploymentValidator.java

```diff
@@ -146,6 +148,22 @@
             return Optional.of("Job could not be upgraded with last-state while HA disabled");
         }
 
+        if (StringUtils.isNullOrWhitespaceOnly(
+                Configuration.fromMap(confMap)
```

Comment:
We could move `Configuration.fromMap(confMap)` to a variable and reuse in this method.